github/resource_github_repository_collaborator.go

```diff
@@ -112,14 +114,14 @@ func resourceGithubRepositoryCollaboratorRead(d *schema.ResourceData, meta inter
 		}
 
 		for _, c := range collaborators {
-			if *c.Login == username {
+			if strings.ToLower(*c.Login) == strings.ToLower(username) {
 				permissionName, err := getRepoPermission(c.Permissions)
 				if err != nil {
 					return err
 				}
 
 				d.Set("repository", repoName)
-				d.Set("username", username)
+				d.Set("username", *c.Login)
```

Comment:
You don't need to deference the pointer here, as the `d.Set` will check that it is a pointer and get the value if the pointer is not nil.